src/client.rs

```
@@ -0,0 +1,613 @@
+            .collect::<Vec<_>>()
+            .chunks(MAX_MUTATIONS_PER_MUTATE_ROWS_REQUEST)
+        {
+            self.multi_set_internal(table_name, chunk.iter().cloned())
+                .await?;
+        }
+        Ok(())
+    }
+
+    async fn multi_set_internal(
+        &mut self,
+        table_name: &str,
+        values: impl IntoIterator<Item = Row> + std::marker::Send,
+    ) -> Result<()> {
+        let entries = values
+            .into_iter()
+            .map(|row| {
+                let mutations = row
+                    .cells
+                    .into_iter()
+                    .map(|cell| Mutation {
+                        mutation: Some(mutation::Mutation::SetCell(SetCell {
+                            family_name: self.column_family.clone(),
+                            column_qualifier: cell.name,
+                            // The timestamp of the cell into which new data should be written.
+                            // Use -1 for current Bigtable server time.
+                            timestamp_micros: -1,
+                            value: cell.value,
+                        })),
+                    })
+                    .collect();
+                Entry {
+                    row_key: row.key,
+                    mutations,
+                }
+            })
+            .collect::<Vec<Entry>>();
+
+        for entries in Self::batch_entries_by_size(entries) {
+            let request = MutateRowsRequest {
+                table_name: self.table_name(table_name),
+                entries,
+                ..MutateRowsRequest::default()
+            };
+
+            let mut response = self.mutate_rows(request).await?;
+            while let Some(part) = response.message().await? {
+                for entry in part.entries {
+                    if let Some(status) = entry.status {
+                        if status.code != 0 {
+                            return Err(BigTableClientError::BigtableWriteError {
```

Should we perhaps collect all errors in a Vec and return an aggregated error instead?
For instance, something like:

```
if !errors.is_empty() {
    return Err(BigTableClientError::BatchMutationErrors(errors));
}
```

This way, we'd clearly see all failed mutations rather than just the first one.

---

@samuel-rufi 

Batch operations should follow "all-or-nothing" semantics. If any mutation in the batch fails, the entire operation should be considered failed. Continuing after errors creates ambiguous state where some mutations succeeded and others didn't.

If we collected errors instead of failing fast, we'd have a scenario where:
- Some chunks succeeded completely
- Some chunks failed partially 
- Some chunks failed completely

This makes error recovery extremely complex.

If you agree we could explore this in a subsequent issue with a dedicated PR 🙏